### PR TITLE
Add support for classification task with 3LC

### PR DESCRIPTION
**Pull request by fredmell**
This PR adds support for the classification task with 3LC by adding a `TLCClassificationTrainer` and a `TLCClassificationValidator`.

Additionally, various other improvements and bugfixes are made. Check out the 3LC Discord for an overview of all the changes.

### CODE REVIEW

File: ultralytics/utils/tlc/classify/utils.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+                    label_column_name=label_column_name,
+                    extensions=IMG_FORMATS,
+                    table_name="original",
+                    dataset_name=f"{name}-{key}",
+                    project_name=f"{name}-YOLOv8", # TODO: Allow user to provide this
+                    if_exists="reuse",
+                    description=f"Original {key} dataset for {data}, created with YOLOv8",
+                )
+
+                table.write_to_url()
```
Comment (gudbrandtandberg) [inline ultralytics/utils/tlc/classify/utils.py:46]: I don't think this is necessary anymore these days.

File: ultralytics/utils/tlc/classify/utils.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+                name = Path(data).name
+
+                table = tlc.Table.from_image_folder(
+                    root=data_dict[key],
+                    image_column_name=image_column_name,
+                    label_column_name=label_column_name,
+                    extensions=IMG_FORMATS,
+                    table_name="original",
+                    dataset_name=f"{name}-{key}",
+                    project_name=f"{name}-YOLOv8", # TODO: Allow user to provide this
```
Comment (gudbrandtandberg) [inline ultralytics/utils/tlc/classify/utils.py:41]: Use self.args.project_name?

File: classify_val_only.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+from ultralytics.utils.tlc.detect.model import TLCYOLO
+from ultralytics.utils.tlc.detect.settings import Settings
+
+model = TLCYOLO("yolov8n-cls.pt")
+
+settings = Settings(
+    image_embeddings_dim=2,
+)
+
+# data + split should be okay
```
Comment (gudbrandtandberg) [inline classify_val_only.py:9]: cleanup

File: ultralytics/utils/tlc/classify/validator.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        dataset = self.build_dataset(dataset_path)
+        return build_dataloader(dataset, batch_size, self.args.workers, shuffle=False, rank=-1)
+
+    def _get_metrics_schemas(self):
+        class_names=[value['internal_name'] for value in self.dataloader.dataset.table.get_value_map(self._label_column_name).values()]
+        column_schemas = {
+            "loss": tlc.Float32Value(),
+            "predicted": tlc.CategoricalLabelSchema(class_names=class_names, display_name="Predicted"), # TODO: Improve the description?
+            "confidence": tlc.Float32Value(value_min=0.0, value_max=1.0),
+            "top1_accuracy": tlc.BoolValue(),
```
Comment (gudbrandtandberg) [inline ultralytics/utils/tlc/classify/validator.py:38]: let's float

File: ultralytics/utils/tlc/detect/settings.py [hunk 5]
```diff
@@ -103,6 +103,25 @@ def verify(self, training: bool = True) -> None:
...
+            return []
+
+        # If start is less than zero, we don't collect during training
+        if self.collection_epoch_start < 0:
+            return []
+
+        if self.collection_epoch_interval <= 0:
+            raise ValueError(f'Invalid interval {self.collection_epoch_interval}, must be non-zero')
+        else:
+            return list(range(self.collection_epoch_start, epochs, self.collection_epoch_interval))
```
Comment (gudbrandtandberg) [inline ultralytics/utils/tlc/detect/settings.py:123]: Would be nice to get printing of these guys up again

File: classify_val_only.py (hunk 3 above)
Comment (fredmell) [reply classify_val_only.py:9]: Agreed to remove these files later. Leaving this up until they are deleted.

File: main.py [hunk 6]
```diff
@@ -1,21 +1,24 @@
...
-model = TLCYOLO("yolov8n.pt")  # initialize
+model = TLCYOLO("yolov8s.pt")  # initialize
 
 # Set 3LC specific settings
 settings = Settings(
-    image_embeddings_dim=2,
-    collection_epoch_start=0,
-    collection_epoch_interval=2,
+    image_embeddings_dim=3,
+    # collection_epoch_start=1,
```
Comment (KellyWalker) [inline main.py:10]: Are these intentionally commented out?

Comment (fredmell) [reply main.py]: Removed the files instead! Perhaps we should have some example scripts under `ultralytics/utils/tlc/examples`.

**Pull request merged**